src/pages/registration/new_registration/LinkRegistrationForm.tsx

```diff
@@ -16,25 +16,39 @@ const StyledTextField = styled(TextField)`
   margin: 0 1rem 0 0;
 `;
 
-export interface DoiFormValues {
+interface DoiFormValues {
   doiUrl: string;
 }
 
 const emptyDoiFormValues: DoiFormValues = {
   doiUrl: '',
 };
 
-const doiUrlPlaceholder = 'https://doi.org/10.1000/xyz123';
+const doiUrlBase = 'https://doi.org/';
+const doiUrlPlaceholder = `${doiUrlBase}10.1000/xyz123`;
+const doiRegExp = new RegExp('\\b(10[.][0-9]{4,}(?:[.][0-9]+)*/(?:(?!["&\'<>])\\S)+)\\b'); // https://stackoverflow.com/a/10324802
 
 interface LinkRegistrationFormProps {
-  handleSearch: (values: { doiUrl: string }) => void;
+  handleSearch: (doiUrl: string) => Promise<void>;
 }
 
 const LinkRegistrationForm = ({ handleSearch }: LinkRegistrationFormProps) => {
   const { t } = useTranslation('registration');
 
+  const onSubmit = async ({ doiUrl }: DoiFormValues, { setValues }: FormikHelpers<DoiFormValues>) => {
+    let newValue = doiUrl.trim().toLowerCase();
+    if (!isValidUrl(newValue)) {
+      const findDoi = doiRegExp.exec(newValue);
+      if (findDoi && findDoi.length > 0) {
+        newValue = `${doiUrlBase}${findDoi[0]}`;
```

Comment:
Dette ser ganske gresk ut for meg. Hva er newValue? Hva er findDoi? Mulig å bruke mer beskrivende variabler? Evt trekke ut til egen funksjon som beskriver funksjonaliteten her?

Reply:
ups, `newValue` henger igjen fra første forsøk hvor dette var i `onChange` (det ble veldig guffent). `regexMatch` er kanskje bedre enn `findDoi`